From 4216897cb08a74dc6081deb8dc0823a061086e34 Mon Sep 17 00:00:00 2001
From: Keunhong Park <keunhong@codeaurora.org>
Date: Thu, 14 Jun 2012 14:44:35 -0600
Subject: [PATCH] Move star icon to header of CommitMessageBlock

Move star icon from the change screen header to header of
CommitMessageBlock so we can remove the screen header.

Change-Id: Ic24be6b7de304d896f84ea77e10799c0e7907793
---
 .../client/changes/ChangeDescriptionBlock.java     |    7 +++--
 .../google/gerrit/client/changes/ChangeScreen.java |    7 +----
 .../gerrit/client/changes/CommitMessageBlock.java  |   27 ++++++++++++++++++++
 .../client/changes/CommitMessageBlock.ui.xml       |    6 +++-
 .../client/changes/PublishCommentScreen.java       |    2 +-
 5 files changed, 38 insertions(+), 11 deletions(-)

diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeDescriptionBlock.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeDescriptionBlock.java
index 9282709..2081040 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeDescriptionBlock.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeDescriptionBlock.java
@@ -19,14 +19,15 @@ import com.google.gerrit.reviewdb.client.Change;
 import com.google.gerrit.reviewdb.client.PatchSetInfo;
 import com.google.gwt.user.client.ui.Composite;
 import com.google.gwt.user.client.ui.HorizontalPanel;
+import com.google.gwtexpui.globalkey.client.KeyCommandSet;
 
 public class ChangeDescriptionBlock extends Composite {
   private final ChangeInfoBlock infoBlock;
   private final CommitMessageBlock messageBlock;
 
-  public ChangeDescriptionBlock() {
+  public ChangeDescriptionBlock(KeyCommandSet keysAction) {
     infoBlock = new ChangeInfoBlock();
-    messageBlock = new CommitMessageBlock();
+    messageBlock = new CommitMessageBlock(keysAction);
 
     final HorizontalPanel hp = new HorizontalPanel();
     hp.add(infoBlock);
@@ -37,6 +38,6 @@ public class ChangeDescriptionBlock extends Composite {
   public void display(final Change chg, final PatchSetInfo info,
       final AccountInfoCache acc) {
     infoBlock.display(chg, acc);
-    messageBlock.display(info.getMessage());
+    messageBlock.display(chg.getId(), info.getMessage());
   }
 }
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeScreen.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeScreen.java
index f761270..289f51a 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeScreen.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ChangeScreen.java
@@ -160,16 +160,11 @@ public class ChangeScreen extends Screen
     keysNavigation.add(new ExpandCollapseDependencySectionKeyCommand(0, 'd', Util.C.expandCollapseDependencies()));
 
     if (Gerrit.isSignedIn()) {
-      StarredChanges.Icon star = StarredChanges.createIcon(changeId, false);
-      star.setStyleName(Gerrit.RESOURCES.css().changeScreenStarIcon());
-      setTitleWest(star);
-
-      keysAction.add(StarredChanges.newKeyCommand(star));
       keysAction.add(new PublishCommentsKeyCommand(0, 'r', Util.C
           .keyPublishComments()));
     }
 
-    descriptionBlock = new ChangeDescriptionBlock();
+    descriptionBlock = new ChangeDescriptionBlock(keysAction);
     add(descriptionBlock);
 
     approvals = new ApprovalTable();
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.java
index 3efe3be..d4fb41e 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.java
@@ -14,7 +14,9 @@
 
 package com.google.gerrit.client.changes;
 
+import com.google.gerrit.client.Gerrit;
 import com.google.gerrit.client.ui.CommentLinkProcessor;
+import com.google.gerrit.reviewdb.client.Change.Id;
 import com.google.gwt.user.client.ui.Composite;
 
 import com.google.gwt.core.client.GWT;
@@ -23,6 +25,8 @@ import com.google.gwt.dom.client.Style.Display;
 import com.google.gwt.uibinder.client.UiBinder;
 import com.google.gwt.uibinder.client.UiField;
 import com.google.gwt.user.client.ui.HTMLPanel;
+import com.google.gwt.user.client.ui.SimplePanel;
+import com.google.gwtexpui.globalkey.client.KeyCommandSet;
 import com.google.gwtexpui.safehtml.client.SafeHtml;
 import com.google.gwtexpui.safehtml.client.SafeHtmlBuilder;
 
@@ -32,6 +36,10 @@ public class CommitMessageBlock extends Composite {
 
   private static Binder uiBinder = GWT.create(Binder.class);
 
+  private KeyCommandSet keysAction;
+
+  @UiField
+  SimplePanel starPanel;
   @UiField
   PreElement commitSummaryPre;
   @UiField
@@ -41,7 +49,26 @@ public class CommitMessageBlock extends Composite {
     initWidget(uiBinder.createAndBindUi(this));
   }
 
+  public CommitMessageBlock(KeyCommandSet keysAction) {
+    this.keysAction = keysAction;
+    initWidget(uiBinder.createAndBindUi(this));
+  }
+
   public void display(final String commitMessage) {
+    display(null, commitMessage);
+  }
+
+  public void display(final Id changeId, final String commitMessage) {
+    if (changeId != null && Gerrit.isSignedIn()) {
+      StarredChanges.Icon star = StarredChanges.createIcon(changeId, false);
+      star.setStyleName(Gerrit.RESOURCES.css().changeScreenStarIcon());
+      starPanel.add(star);
+
+      if (keysAction != null) {
+        keysAction.add(StarredChanges.newKeyCommand(star));
+      }
+    }
+
     String[] splitCommitMessage = commitMessage.split("\n", 2);
 
     String commitSummary = splitCommitMessage[0];
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.ui.xml b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.ui.xml
index a0fcf0a..2e434a6 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.ui.xml
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CommitMessageBlock.ui.xml
@@ -62,11 +62,15 @@ limitations under the License.
     .commitBody {
       margin-top: 10px;
     }
+
+    .starPanel {
+      float: left;
+    }
   </ui:style>
 
   <g:HTMLPanel>
     <table class='{style.commitMessageTable}'>
-      <tr><td class='{style.header}'>Commit Message</td></tr>
+      <tr><td class='{style.header}'><g:SimplePanel styleName='{style.starPanel}' ui:field='starPanel'></g:SimplePanel>Commit Message</td></tr>
       <tr><td class='{style.contents}'>
         <pre class='{style.commitSummary} {res.css.changeScreenDescription}' ui:field='commitSummaryPre'/>
         <pre class='{style.commitBody} {res.css.changeScreenDescription}' ui:field='commitBodyPre'/>
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/PublishCommentScreen.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/PublishCommentScreen.java
index 0c08491..6c6a0ab 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/PublishCommentScreen.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/PublishCommentScreen.java
@@ -83,7 +83,7 @@ public class PublishCommentScreen extends AccountScreen implements
     addStyleName(Gerrit.RESOURCES.css().publishCommentsScreen());
 
     approvalButtons = new ArrayList<ValueRadioButton>();
-    descBlock = new ChangeDescriptionBlock();
+    descBlock = new ChangeDescriptionBlock(null);
     add(descBlock);
 
     final FormPanel form = new FormPanel();
-- 
1.7.8.3

